packages/astro/src/core/render/params-and-props.ts

```diff
@@ -77,7 +78,10 @@ export function getParams(route: RouteData, pathname: string): Params {
 	if (!route.params.length) return {};
 	// The RegExp pattern expects a decoded string, but the pathname is encoded
 	// when the URL contains non-English characters.
-	const paramsMatch = route.pattern.exec(pathname);
+	const paramsMatch =
+		route.pattern.exec(pathname) ||
+		route.fallbackRoutes.map((fallbackRoute) => fallbackRoute.pattern.exec(pathname))[0];
```

Comment:
This doesn't seem correct to me. We're looping over all fallback routes and use `exec`, but we always pick the first one. Instead, we should use a for loop, and assign to `paramsMatch` that matches the route.

```js
for (const fallbackRoute in route.fallbackRoutes) {
	if (fallbackRoute.test(pathname)) {
		paramsMatch = fallbackRoute.exec(pathname);
		break;
	}
}
```

Reply:
updated to `.find(x->x)` to get first non-null match instead of just first, which could be null